### PR TITLE
TRAN-139: server side pagination, refactor form compoments, apiClient refactor

### PR DESCRIPTION
**Pull request by sniedzielski**
TICKET: https://sd-transit.atlassian.net/browse/TRAN-139

major changes regarding server-side pagination:
- added additional config for React-Table to use Types in TypeScript for this component. (react-table-config.d.ts)
- server side pagination mechanism implemented in currently implemented pages
- server side pagination mechanism implemented in Table component.
- changed the way of processing GET request because of using this new approach (remove GET dispatch from Searcher and on 'init page')  
- default offset pagination set into 10 (related PR on backend - https://github.com/SD-TransIT/transit_backend/pull/25). In the future it could be possible

Additional changes
- refactor of API client
- refactor of Manual Upload forms - used 'DeleteModal' and FormHeader components
- added red borders on LoginForm when there are errors on particular field of LoginForm

Example of working pagination:
Peek 2022-09-12 08-40

LoginForm:
Screenshot from 2022-09-12 08-45-16

### CODE REVIEW

File: src/pages/menuUpload/itemMaster/ItemMasterPage.tsx [hunk 1]
```diff
@@ -30,28 +33,77 @@ import itemColumns from './columnsItem';
 const clearValues: IItem = { id: undefined, name: '', conditions: '' };
 
 function ItemMasterPage() {
+  const DEFAULT_OFFSET = 10;
+  const FIRST_PAGE = 1;
+  const EMPTY_SEARCHER = '';
+  const itemUrl = 'item/';
```
Comment (KarolinaKopacz) [inline src/pages/menuUpload/itemMaster/ItemMasterPage.tsx]: Do we want to have const values inside the function or outside like `clearValues`?

File: src/pages/menuUpload/supplierMaster/SupplierMasterPage.tsx [hunk 2]
```diff
@@ -16,42 +18,92 @@ import AddItemButton from 'shared/buttons/AddItemButton';
...
 } from 'stores/types/supplierMasterType';
+import { getRequest } from 'utils/apiClient';
 
 const clearValues: ISupplierMaster = { id: undefined, name: '' };
 
 function SupplierMasterPage() {
+  const DEFAULT_OFFSET = 10;
+  const FIRST_PAGE = 1;
+  const EMPTY_SEARCHER = '';
+  const supplierUrl = 'supplier/';
```
Comment (KarolinaKopacz) [inline src/pages/menuUpload/supplierMaster/SupplierMasterPage.tsx]: This one can be exported from `stores/reducers/supplierMasterReducer.ts` - if url will change we will need to change it only in one place

**Review by KarolinaKopacz — CHANGES_REQUESTED** [review]

File: src/pages/menuUpload/itemMaster/ItemMasterPage.tsx (hunk 1 above)
Comment (sniedzielski) [reply src/pages/menuUpload/itemMaster/ItemMasterPage.tsx]: good idea. I can change this

File: src/pages/menuUpload/supplierMaster/SupplierMasterPage.tsx (hunk 2 above)
Comment (sniedzielski) [reply src/pages/menuUpload/supplierMaster/SupplierMasterPage.tsx]: good catch. I will change this

File: src/pages/menuUpload/customerType/columnsCustomerType.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+const customerTypeColumns = [
+  {
+    Header: 'Id',
+    accessor: 'id',
```
Comment (KarolinaKopacz) [inline src/pages/menuUpload/customerType/columnsCustomerType.ts:4]: Can you also add here width? :)

**Review by KarolinaKopacz — CHANGES_REQUESTED** [review]

File: src/pages/menuUpload/customerType/columnsCustomerType.ts (hunk 3 above)
Comment (sniedzielski) [reply src/pages/menuUpload/customerType/columnsCustomerType.ts:4]: yes, of course

**Comment by sniedzielski** [comment]
fixed @KarolinaKopacz . I moved some const to utils/const.ts. Moreover urls to backend on pages are now exported from sagas.

**Review by KarolinaKopacz — APPROVED** [review]

**Pull request merged**